### PR TITLE
Add co2 calc trip details

### PR DESCRIPTION
**Pull request by daniel-heppner-ibigroup**
This adds CO2 value to trip details. Core-utils update appears to have already made it into master.

### CODE REVIEW

File: packages/trip-details/src/types.ts
Comment (miles-grant-ibigroup) [inline packages/trip-details/src/types.ts:76]: This comment looks like it might be outdated

File: packages/trip-details/src/types.ts
Comment (miles-grant-ibigroup) [inline packages/trip-details/src/types.ts:15]: I would make this optional since it'll still be false if missing

File: packages/core-utils/src/itinerary.js
Comment (miles-grant-ibigroup) [inline packages/core-utils/src/itinerary.js:491]: This PR is blocked until the other one is merged right? so this change goes away

File: packages/trip-details/src/index.tsx
Comment (miles-grant-ibigroup) [inline packages/trip-details/src/index.tsx:342]: Do we want to add an info button that links to the resource we used to generate the values? Or maybe just explains the formula or something?

**Review by miles-grant-ibigroup — APPROVED** [review]
This looks really cool thank you! 

Just a few small touchups and I think we should be good to go!

File: packages/trip-details/src/index.tsx
Comment (daniel-heppner-ibigroup) [reply packages/trip-details/src/index.tsx:342]: Added!

File: packages/trip-details/src/index.tsx
Comment (miles-grant-ibigroup) [inline packages/trip-details/src/index.tsx:341]: You might want to throw a ? after co2config. Getting some white screen of death errors

File: packages/trip-details/src/index.tsx
Comment (miles-grant-ibigroup) [inline packages/trip-details/src/index.tsx:371]: Could we change this to include some of the values we use? For example "CO2 output is calculated by multiplying the distance of the trip (2.4km) by the average CO2 emitted by each mode involved."

**Review by miles-grant-ibigroup — CHANGES_REQUESTED** [review]
Having some issues testing this from within otp-rr (because of the separate core-utils) so I would say let's leave this blocked until that is through.

File: packages/trip-details/src/index.tsx
Comment (daniel-heppner-ibigroup) [reply packages/trip-details/src/index.tsx:371]: I honestly don't think that would be all that useful information, and it isn't that simple to implement. I don't have immediate access to the trip distance (but I could easily calculate it with a reducer on legs). More importantly, unless we always displayed km, this would also involve unit conversion. And I don't see it as being very useful, since at no point in the calculation are we using the total trip distance (since it's calculated per-leg).

Comment (miles-grant-ibigroup) [reply packages/trip-details/src/index.tsx:371]: That makes sense! Definitely a nice-to-have so if it's a ton of effort to implement probably not worth it.

Do you think linking to our average emission info is a good idea?

Comment (daniel-heppner-ibigroup) [reply packages/trip-details/src/index.tsx:371]: What do you mean by linking to our average emissions?

Comment (miles-grant-ibigroup) [reply packages/trip-details/src/index.tsx:371]: We should probably link to the pdf we used to get the values we use to calculate the emissions the same way we do for the calories burned!

**Comment by demory** [comment]
With this code checked out (including the corresponding branches in otp-rr and configurations), I get the following error after planning a trip and dragging a start/end marker on the map:

image

**Comment by demory** [comment]
A couple accessibility related notes:
- I cannot get a screenreader (using Mac VoiceOver w/ Chrome) to read the "Trip Details" box, including the CO2 info, beyond the header. It reads "Trip Details" but then jumps back to the itinerary summary. Can anyone reproduce?
- The blue "?" buttons are just read as "what does this mean" for both the fare and CO2 items. These should be more specific to each one, maybe "Show fare details" / "Show CO2 details" or similar?

**Comment by daniel-heppner-ibigroup** [comment]
@binh-dam-ibigroup could you help out with some translations?

File: packages/trip-details/src/index.tsx
Comment (daniel-heppner-ibigroup) [reply packages/trip-details/src/index.tsx:328]: And added a link to the spreadsheet containing the formula.

File: packages/trip-details/src/index.tsx
Comment (daniel-heppner-ibigroup) [reply packages/trip-details/src/index.tsx:361]: Link added.

File: packages/trip-details/src/index.tsx
Comment (miles-grant-ibigroup) [inline packages/trip-details/src/index.tsx:50]: Do we not want the itf to know we're using their report?

**Review by miles-grant-ibigroup — APPROVED** [review]
I know this has been a long work in progress but it looks and works great! Can't wait to see the full otp-rr PR!

File: packages/trip-details/src/index.tsx
Comment (daniel-heppner-ibigroup) [reply packages/trip-details/src/index.tsx:50]: ESlint wants those on there for security reasons if your target is a new window... I don't ask questions.

Comment (daniel-heppner-ibigroup) [reply packages/trip-details/src/index.tsx:50]: ESlint wants those on there for security reasons if your target is a new window... I don't ask questions.

File: packages/trip-details/i18n/fr.yml
Comment (demory) [inline packages/trip-details/i18n/fr.yml:11]: Should this be translated?

Comment (binh-dam-ibigroup) [reply packages/trip-details/i18n/fr.yml:11]: Done in https://github.com/opentripplanner/otp-ui/pull/345/commits/1afba88c57b910c63040c7eed23e5b5ff64b1a6a.

**Review by demory — APPROVED** [review]

**Pull request merged**
